File: backend/lambda/shared/python/shared/isbn.py
import re


def normalize_isbn(raw_value: str) -> str | None:
    digits = re.sub(r"[^0-9Xx]", "", raw_value).upper()

    if len(digits) == 13 and (digits.startswith("978") or digits.startswith("979")):
        return digits

    if len(digits) == 10:
        return digits

    return None

File: frontend/src/lib/isbn.ts
export const normalizeIsbn = (rawValue: string): string | null => {
  const digits = rawValue.replace(/[^0-9X]/gi, "").toUpperCase();

  if (digits.length === 13 && (digits.startsWith("978") || digits.startsWith("979"))) {
    return digits;
  }

  if (digits.length === 10) {
    return digits;
  }

  return null;
};
